csharp/lib/src/lib.rs

```diff
@@ -118,7 +119,11 @@ pub extern "C" fn set(
             let client = Box::leak(Box::from_raw(ptr_address as *mut Client));
             match result {
                 Ok(_) => (client.success_callback)(callback_index, std::ptr::null()), // TODO - should return "OK" string.
-                Err(_) => (client.failure_callback)(callback_index), // TODO - report errors
+                Err(err) => {
+                    logger_core::log_debug("command error", format!("callback {}, error {}, kind {:?}, code {:?}, category {:?}, detail {:?}", callback_index, err, err.kind(), err.code(), err.category(), err.detail()));
```

Comment:
Doesn't err implement Debug or Display? I think we should be able to just print it without explicitly extracting the kind, code, category, and detail.

Reply:
Fixed in 93186f3e